flow/Arena.cpp

```diff
@@ -438,7 +438,7 @@ void testIteratorIncrement() {
 			i += 2;
 		}
 	}
-	{
+	if (size > 0) {
```

Comment:
What exactly goes wrong? I agree that `xs.end() - 1` is not legitimate if the container is empty, but I'm curious how this actually causes the test to fail.

Reply:
The failure is at the assertion below, i.e., `*iter`:
`444                             ASSERT(*iter == StringRef(std::to_string(i)));`

Reply:
Interesting. I'd have thought that `xs.end() - 1` would not be `>= xs.begin()`

Reply:
Yes, this an interesting one. I guess `xs.begin()` is 0, while `iter` is `0xfffffffffffffff4`.

```
(gdb) p iter
$20 = (StringRef *) 0xfffffffffffffff4
(gdb) p xs
$21 = {
  <> = {
    static composed_identifier_offset = 24, 
    static composition_depth = <optimized out>, 
    static file_identifier = 14065384
  }, 
  <> = {<No data fields>}, 
  members of VectorRef<StringRef, VecSerStrategy::FlatBuffers>: 
  data = 0x0, 
  m_size = 0, 
  m_capacity = 0
}

```